src/coord/src/session/vars.rs

```diff
@@ -676,3 +704,104 @@ impl Value for [&str] {
+    Debug2,
+    /// Sends all messages to the client, since all DEBUG levels are treated as the same right now.
+    Debug3,
+    /// Sends all messages to the client, since all DEBUG levels are treated as the same right now.
+    Debug4,
+    /// Sends all messages to the client, since all DEBUG levels are treated as the same right now.
+    Debug5,
+    /// Sends only NOTICE, WARNING, INFO, ERROR, FATAL and PANIC level messages.
+    /// Not listed as a valid value, but accepted by Postgres
+    Info,
+}
+
+impl ClientSeverity {
+    fn as_str(&self) -> &'static str {
+        match self {
+            ClientSeverity::Error => "error",
+            ClientSeverity::Warning => "warning",
+            ClientSeverity::Notice => "notice",
+            ClientSeverity::Info => "info",
+            ClientSeverity::Log => "log",
+            ClientSeverity::Debug1 => "debug1",
+            ClientSeverity::Debug2 => "debug2",
+            ClientSeverity::Debug3 => "debug3",
+            ClientSeverity::Debug4 => "debug4",
+            ClientSeverity::Debug5 => "debug5",
+        }
+    }
+
+    fn valid_values() -> Vec<&'static str> {
+        // INFO left intentionally out, to match Postgres
+        vec![
+            ClientSeverity::Debug5.as_str(),
+            ClientSeverity::Debug4.as_str(),
+            ClientSeverity::Debug3.as_str(),
+            ClientSeverity::Debug2.as_str(),
+            ClientSeverity::Debug1.as_str(),
+            ClientSeverity::Log.as_str(),
+            ClientSeverity::Notice.as_str(),
+            ClientSeverity::Warning.as_str(),
+            ClientSeverity::Error.as_str(),
+        ]
+    }
+}
+
+impl Value for ClientSeverity {
+    const TYPE_NAME: &'static str = "string";
+
+    fn parse(s: &str) -> Result<Self::Owned, ()> {
+        let s = UncasedStr::new(s);
+
+        if s == ClientSeverity::Error.as_str() {
+            Ok(ClientSeverity::Error)
+        } else if s == ClientSeverity::Warning.as_str() {
+            Ok(ClientSeverity::Warning)
+        } else if s == ClientSeverity::Notice.as_str() {
+            Ok(ClientSeverity::Notice)
+        } else if s == ClientSeverity::Info.as_str() {
+            Ok(ClientSeverity::Info)
+        } else if s == ClientSeverity::Log.as_str() {
+            Ok(ClientSeverity::Log)
+        } else if s == ClientSeverity::Debug1.as_str() {
+            Ok(ClientSeverity::Debug1)
+        // Postgres treats `debug` as an input as equivalent to `debug2`
+        } else if s == ClientSeverity::Debug2.as_str() || s == "debug" {
+            Ok(ClientSeverity::Debug2)
+        } else if s == ClientSeverity::Debug3.as_str() {
+            Ok(ClientSeverity::Debug3)
+        } else if s == ClientSeverity::Debug4.as_str() {
+            Ok(ClientSeverity::Debug4)
+        } else if s == ClientSeverity::Debug5.as_str() {
+            Ok(ClientSeverity::Debug5)
+        } else {
+            Err(())
+        }
+    }
+
+    fn format(&self) -> String {
```

Comment:
Is this called anywhere?

Reply:
Yes, this is what is used when you run `SHOW $varname` and similar, as well as error messages.